src/services/wallet-service/wallet-service.spec.ts

```diff
@@ -0,0 +1,201 @@
+        walletService = new WalletService(storage, <MerkleTreeService>merkleTreeService);
+    });
+
+    it("should return an empty array when there are no wallets saved", (done) => {
+        spyOn(storage, "get").and.returnValue(Promise.resolve());
+        walletService.getAll().then(data => {
+            expect(storage.get).toHaveBeenCalledWith(WALLET_STORAGE_KEY);
+            expect(data).toEqual([]);
+            done();
+        });
+    });
+
+    it("should return three specific wallets when retrieving the wallets from storage", (done) => {
+        mockedWalletService.getAll().then(data => {
+            spyOn(storage, "get").and.returnValue(Promise.resolve(data));
+            walletService.getAll().then(data => {
+                expect(data).toEqual(<any>[
+                    {"id":"012d294e-cb11-439b-937a-12d47a52c305",
+                        "type":"local",
+                        "name":"Biosta",
+                        "publicKey":
+                        "ETm9QUJLVdJkTqRojTNqswmeAQGaofojJJ",
+                        "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"},
+                    {"id":"9b5329ff-c683-42a5-9165-4093e4076166",
+                        "type":"local",
+                        "name":"Labilo",
+                        "publicKey":"ELsKCchf9rcGsufjRR62PG5Fn5dFinfgeN",
+                        "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"},
+                    {"id":"a2e16167-fedb-47d2-8856-2b3f97389c35",
+                        "type":"local",
+                        "name":"Zalista",
+                        "publicKey":"EZ7tP3CBdBKrB9MaBgZNHyDcTg5TFRRpaY",
+                        "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"}]);
+                done();
+            });
+        });
+    });
+
+    it("should return cloned wallets when there are already existing wallets", () => {
+        spyOn(storage, "get").and.callThrough();
+        (<any>walletService).wallets = [
+            {"id":"012d294e-cb11-439b-937a-12d47a52c305",
+                "type":"local",
+                "name":"Biosta",
+                "publicKey":
+                "ETm9QUJLVdJkTqRojTNqswmeAQGaofojJJ",
+                "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"}];
+        walletService.getAll();
+        expect(storage.get).not.toHaveBeenCalled();
+    });
+
+    it("should push a new wallet when it does not yet exist and overwrite when it already exists", (done) => {
+        // So that getWalletIndex doesn't crash on undefined length
+        (<any>walletService).wallets = [];
+        // Let to reset later
+        let pushSpy = spyOn((<any>walletService).wallets, "push").and.callThrough();
+        spyOn(storage, "set").and.callThrough();
+        let wallet = <IWallet>{id : "4b6cff11-5888-43bb-bde1-911e12b659e6",
```

Comment:
Instead of casting you can also write this like this:
```
let wallet: IWallet = {...}
```

Of course this only works if the object is truly a valid IWallet